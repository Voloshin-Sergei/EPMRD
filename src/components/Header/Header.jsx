import React from 'react';
import { Search } from '../Search';

import style from './Header.module.scss';

export const Header = () => (
  <header className={style.header}>
    <p className={style.title}>Movieroulette</p>
    <Search />
  </header>
);
