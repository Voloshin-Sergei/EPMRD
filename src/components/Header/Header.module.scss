@import 'Styles/variables/colors';
@import 'Styles/variables/fonts';
@import 'Styles/variables/responsive';

.header {
  background: linear-gradient(rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0.8)),
    url('../../assets/header_bg.jpg') no-repeat center / cover;
  padding: 2rem 5rem;

  @media screen and (max-width: $mobile--sm) {
    padding: 2rem 2rem;
  }
}

.title {
  margin-bottom: 8rem;
  font-size: 3.6rem;
  font-weight: $font-weight--medium;
  text-transform: uppercase;
  color: $color-secondary;

  @media screen and (max-width: $mobile) {
    margin-bottom: 4rem;
  }

  @media screen and (max-width: $mobile--sm) {
    margin-bottom: 2rem;
  }
}
