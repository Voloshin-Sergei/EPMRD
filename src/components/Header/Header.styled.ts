import styled from 'styled-components';
import { colors } from '../../styles/variables/colors.styled';
import { breakpoints } from '../../styles/variables/responsive.styled';
import { fonts } from '../../styles/variables/fonts.styled';
import bgImg from '../../assets/header_bg.jpg';

export const StyledHeader = styled.header`
  background: linear-gradient(${colors.colorPrimaryShade}, ${colors.colorPrimaryShade}),
    url(${bgImg}) no-repeat center / cover;
`;
export const StyledTitle = styled.p`
  margin: 2rem 0 8rem 5rem;
  color: ${colors.colorSecondary};
  font-weight: ${fonts.fontWeightMedium};
  font-size: 3.6rem;
  text-transform: uppercase;

  @media screen and (max-width: ${breakpoints.mobile}) {
    margin-bottom: 4rem;
  }

  @media screen and (max-width: ${breakpoints.mobileSm}) {
    margin-bottom: 2rem;
  }
`;
