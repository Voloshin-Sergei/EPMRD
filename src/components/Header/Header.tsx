import React, { ReactNode } from 'react';

import { StyledHeader } from './Header.styled';

export const Header: React.FC<ReactNode> = ({ children }) => {
  return <StyledHeader>{children}</StyledHeader>;
};
