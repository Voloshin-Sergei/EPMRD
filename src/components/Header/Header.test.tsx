import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Header }  from './Header';
Enzyme.configure({ adapter: new Adapter() });

describe("Header component", () => {
    it('should render Header component', () => {
    const component = render(<Header/>); 
    expect(component).toMatchSnapshot();
  });
});