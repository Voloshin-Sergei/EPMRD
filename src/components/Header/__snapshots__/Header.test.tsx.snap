// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
<header
  class="header"
>
  <p
    class="title"
  >
    Movieroulette
  </p>
  <p
    class="title"
  >
    find your movie
  </p>
  <form
    class="form"
  >
    <input
      class="field"
      placeholder="search movie"
      type="text"
    />
    <div
      class="buttons"
    >
      <span
        class="text"
      >
        search by
      </span>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <button
            class="button tag primary"
            type="button"
          >
            title
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="button tag primary"
            type="button"
          >
            genre
          </button>
        </li>
      </ul>
      <button
        class="button search primary"
        type="button"
      >
        search
      </button>
    </div>
  </form>
</header>
`;
